src/main/cms/cms_menu_imu.c

```diff
@@ -0,0 +1,278 @@
+    }
+    tempPid[3][0] = masterConfig.profile[profileIndex].pidProfile.P8[PIDLEVEL];
+    tempPid[3][1] = masterConfig.profile[profileIndex].pidProfile.I8[PIDLEVEL];
+    tempPid[3][2] = masterConfig.profile[profileIndex].pidProfile.D8[PIDLEVEL];
+
+    return 0;
+}
+
+static long cmsx_PidWriteback(OSD_Entry *self)
+{
+    UNUSED(self);
+
+    for (uint8_t i = 0; i < 3; i++) {
+        masterConfig.profile[profileIndex].pidProfile.P8[i] = tempPid[i][0];
+        masterConfig.profile[profileIndex].pidProfile.I8[i] = tempPid[i][1];
+        masterConfig.profile[profileIndex].pidProfile.D8[i] = tempPid[i][2];
+    }
+
+    masterConfig.profile[profileIndex].pidProfile.P8[PIDLEVEL] = tempPid[3][0];
+    masterConfig.profile[profileIndex].pidProfile.I8[PIDLEVEL] = tempPid[3][1];
+    masterConfig.profile[profileIndex].pidProfile.D8[PIDLEVEL] = tempPid[3][2];
+
+    return 0;
+}
+
+static OSD_Entry cmsx_menuPidEntries[] =
+{
+    { "-- PID --", OME_Label, NULL, NULL, 0},
+
+    { "ROLL  P", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDROLL][0],  0, 200, 1 }, 0 },
+    { "ROLL  I", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDROLL][1],  0, 200, 1 }, 0 },
+    { "ROLL  D", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDROLL][2],  0, 200, 1 }, 0 },
+
+    { "PITCH P", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDPITCH][0], 0, 200, 1 }, 0 },
+    { "PITCH I", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDPITCH][1], 0, 200, 1 }, 0 },
+    { "PITCH D", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDPITCH][2], 0, 200, 1 }, 0 },
+
+    { "YAW   P", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDYAW][0],   0, 200, 1 }, 0 },
+    { "YAW   I", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDYAW][1],   0, 200, 1 }, 0 },
+    { "YAW   D", OME_UINT8, NULL, &(OSD_UINT8_t){ &tempPid[PIDYAW][2],   0, 200, 1 }, 0 },
+
+    { "BACK", OME_Back, NULL, NULL, 0 },
+    { NULL, OME_END, NULL, NULL, 0 }
+};
+
+static CMS_Menu cmsx_menuPid = {
+    "XPID",
+    OME_MENU,
+    cmsx_PidRead,
+    cmsx_PidWriteback,
+    NULL,
+    cmsx_menuPidEntries,
+};
+
+//
+// Rate & Expo
+//
+static controlRateConfig_t rateProfile;
+
+static long cmsx_RateExpoRead(void)
+{
+    memcpy(&rateProfile, &masterConfig.profile[masterConfig.current_profile_index].controlRateProfile[masterConfig.profile[masterConfig.current_profile_index].activeRateProfile], sizeof(controlRateConfig_t));
+
+    return 0;
+}
+
+static long cmsx_RateExpoWriteback(OSD_Entry *self)
+{
+    UNUSED(self);
+
+    memcpy(&masterConfig.profile[masterConfig.current_profile_index].controlRateProfile[masterConfig.profile[masterConfig.current_profile_index].activeRateProfile], &rateProfile, sizeof(controlRateConfig_t));
+
+    return 0;
+}
+
+static long cmsx_menuRcConfirmBack(OSD_Entry *self)
+{
+    if (self && self->type == OME_Back)
+        return 0;
+    else
+        return -1;
+}
+
+static OSD_Entry cmsx_menuRateExpoEntries[] =
+{
+    { "-- RATE&EXPO --", OME_Label, NULL, NULL, 0 },
```

Comment:
Rate profile number should be shown on this screen.

Also, it might be beneficial to allow for rate profile / profile switching through the CMS, so users get to configure all settings.